File: test-suite-python/src/test/java/io/micronaut/docs/function/aws/test/SampleRequestHandler.java
package io.micronaut.docs.function.aws.test;

// An AWS Lambda handler must extend the Java class MicronautRequestHandler, which a Python class cannot do, so the
// handler tested by the Python RequestHandlerTest is this Java copy of the test-suite SampleRequestHandler.

//tag::clazz[]
import io.micronaut.context.ApplicationContext;
import io.micronaut.function.aws.MicronautRequestHandler;

public class SampleRequestHandler extends MicronautRequestHandler<String, String> {
    // Used in AWS
    public SampleRequestHandler() {
    }

    // Used in tests
    public SampleRequestHandler(ApplicationContext applicationContext) {
        super(applicationContext);
    }

    @Override
    public String execute(String input) {
        return "Hello " + input;
    }
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/function/aws/test/SampleRequestHandler.py
# tag::clazz[]
from micronaut.context import ApplicationContext
from micronaut.function.aws import MicronautRequestHandler


class SampleRequestHandler(MicronautRequestHandler[str, str]):

    # Used in tests
    def __init__(self, application_context: ApplicationContext):
        super().__init__(application_context)

    def execute(self, input: str) -> str:
        return "Hello " + input
# end::clazz[]
